### PR TITLE
Analysis fixes

### PR DESCRIPTION
**Pull request by attilamarx**
fixed issue: #1
modified:
code/greedy_algorithms/src/activity_selection/ActivitySelection.java -- +1 if statement instead of the nested ? :
code/greedy_algorithms/src/dijkstra_shortest_path/Dijkstra.java -- private constructor
code/greedy_algorithms/src/fractional_knapsack/FractionalKnapsack.java -- double casting
code/greedy_algorithms/src/hillclimber/Hillclimber.java -- removed unnecessary local variable
code/greedy_algorithms/src/minimum_coins/MinimumCoins.java -- public attrubutes to final

### CODE REVIEW

File: code/greedy_algorithms/src/fractional_knapsack/FractionalKnapsack.java [hunk 1]
```diff
@@ -45,8 +45,8 @@ public int getRemainingWeight() {
         
         if( i < itemList.length) {
             //Calculating the fraction of the item whoes weight is greater than the current remaining weight
-            finalValue = finalValue + (remainingWeight) * itemList[i].getValue() / itemList[i].getWeight();
-            itemList[i].setFraction(remainingWeight / itemList[i].getWeight());
+            finalValue = finalValue + (remainingWeight) * itemList[i].getValue() / (double)itemList[i].getWeight();
```
Comment (Bergrei) [inline code/greedy_algorithms/src/fractional_knapsack/FractionalKnapsack.java]: Did you check if it was an intentional int devide to get the floor of the result?

File: code/greedy_algorithms/src/activity_selection/ActivitySelection.java [hunk 2]
```diff
@@ -57,6 +57,8 @@ public String toString() {
 class ActivityComparator implements Comparator<Activity> {
     @Override
     public int compare(Activity a, Activity b) {
-        return a.finish < b.finish ? -1 : a.finish == b.finish ? 0 : 1;
+		if (a.finish < b.finish)
+			return -1;
+		return a.finish == b.finish ? 0 : 1;
```
Comment (Bergrei) [inline code/greedy_algorithms/src/activity_selection/ActivitySelection.java]: I would consider separating this statement too, but already is much more readable.

Comment (Bergrei) [reply code/greedy_algorithms/src/activity_selection/ActivitySelection.java]: I meant the line with the return, but it seem github ignores the specific line of the comment.

Comment (attilamarx) [reply code/greedy_algorithms/src/activity_selection/ActivitySelection.java]: I really don't think its necessary and neither does SonarQube ( ͡° ͜ʖ ͡°)

**Comment by attilamarx** [comment]
Travis All checks have failed... Seems good 10/10 would merge, so will I!

**Pull request merged**